src/main/java/org/team1540/robot2023/RobotContainer.java

```diff
@@ -58,45 +59,36 @@ public RobotContainer() {
+        controlPanel.onButton(ButtonPanel.PanelButton.TOP_CENTER   ).whileTrue(new GridDriveAndPivotCommand(drivetrain, PolePosition.CENTER, arm, Constants.Auto.angleHighCube));
+        controlPanel.onButton(ButtonPanel.PanelButton.TOP_RIGHT    ).whileTrue(new GridDriveAndPivotCommand(drivetrain, PolePosition.RIGHT,  arm, Constants.Auto.angleHighCone));
+        controlPanel.onButton(ButtonPanel.PanelButton.MIDDLE_LEFT  ).whileTrue(new GridDriveAndPivotCommand(drivetrain, PolePosition.LEFT,   arm, Constants.Auto.angleMidCone));
+        controlPanel.onButton(ButtonPanel.PanelButton.MIDDLE_CENTER).whileTrue(new GridDriveAndPivotCommand(drivetrain, PolePosition.CENTER, arm, Constants.Auto.angleMidCube));
+        controlPanel.onButton(ButtonPanel.PanelButton.MIDDLE_RIGHT ).whileTrue(new GridDriveAndPivotCommand(drivetrain, PolePosition.RIGHT,  arm, Constants.Auto.angleMidCone));
+        controlPanel.onButton(ButtonPanel.PanelButton.BOTTOM_LEFT  ).whileTrue(new GridDriveAndPivotCommand(drivetrain, PolePosition.LEFT,   arm, Constants.Auto.angleDown));
+        controlPanel.onButton(ButtonPanel.PanelButton.BOTTOM_CENTER).whileTrue(new GridDriveAndPivotCommand(drivetrain, PolePosition.CENTER, arm, Constants.Auto.angleDown));
+        controlPanel.onButton(ButtonPanel.PanelButton.BOTTOM_RIGHT ).whileTrue(new GridDriveAndPivotCommand(drivetrain, PolePosition.RIGHT,  arm, Constants.Auto.angleDown));
         copilot.a().toggleOnTrue(new GrabberIntakeCommand(wheeledGrabber));
-        // coop:button(B,Run Outtake [HOLD],copilot)
         copilot.b().whileTrue(new GrabberOuttakeCommand(wheeledGrabber));
-
-
-
         //Pneumatic Control
 //        copilot.a().onTrue(new InstantCommand(() -> pneumaticClaw.toggle()));
 //        copilot.a().onTrue(new InstantCommand(() -> pneumaticClaw.set(true)));
 //        copilot.b().onTrue(new InstantCommand(() -> pneumaticClaw.set(false)));
-
-
-        copilot.x().whileTrue(new ExtensionCommand(arm, 30));
-        copilot.y().whileTrue(new ExtensionCommand(arm, 40));
-        // coop:button(RBumper, Move to arm straight forward [HOLD],copilot)
-        copilot.rightBumper().whileTrue(new PivotToSetpoint(arm, Rotation2d.fromDegrees(90)));
-        // coop:button(LBumper, Move to arm straight up [HOLD],copilot)
-        copilot.leftBumper().whileTrue(new PivotToSetpoint(arm, Rotation2d.fromDegrees(0)));
-        copilot.leftStick().onTrue(new InstantCommand(() -> arm.resetAngle())); // TODO: 2/18/2023 change binding
-
+        //Ground pickup and hybrid node deposit position
+        copilot.x().whileTrue(new RetractAndPivotCommand(arm, Rotation2d.fromDegrees(-115)));
+        //Upright position for driving
+        copilot.leftBumper().whileTrue(new RetractAndPivotCommand(arm, Rotation2d.fromDegrees(0)));
+        //Substation pickup, extension is needed
+        copilot.rightBumper().whileTrue(new RetractAndPivotCommand(arm, Rotation2d.fromDegrees(-58)));
```

Comment:
Since we now have an end condition for RetractAndPivotCommand, these should probably just be onTrue rather than whileTrue

Reply:
I disagree here, I'd always like the copilot to be able to stop pressing and have the robot stop moving